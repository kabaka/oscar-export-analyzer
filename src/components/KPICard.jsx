import React from 'react';

/**
 * Card component displaying a key performance indicator (KPI) with optional sparkline.
 */
export default function KPICard({ title, value, children }) {
  return (
    <div className="kpi-card">
      <div className="title">{title}</div>
      <div className="value">{value}</div>
      {children}
    </div>
  );
}
