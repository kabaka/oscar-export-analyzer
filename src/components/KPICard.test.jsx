import React from 'react';
import { render, screen } from '@testing-library/react';
import KPICard from './KPICard';

describe('KPICard', () => {
  it('renders title, value, and children', () => {
    const { asFragment } = render(
      <KPICard title="Test KPI" value="42">
        <span data-testid="child">Child Content</span>
      </KPICard>,
    );
    expect(screen.getByText('Test KPI')).toBeInTheDocument();
    expect(screen.getByText('42')).toBeInTheDocument();
    expect(screen.getByTestId('child')).toHaveTextContent('Child Content');
    expect(asFragment()).toMatchSnapshot();
  });
});
