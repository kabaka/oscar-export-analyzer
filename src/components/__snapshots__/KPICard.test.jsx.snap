// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`KPICard > renders title, value, and children 1`] = `
<DocumentFragment>
  <div
    class="kpi-card"
  >
    <div
      class="title"
    >
      Test KPI
    </div>
    <div
      class="value"
    >
      42
    </div>
    <span
      data-testid="child"
    >
      Child Content
    </span>
  </div>
</DocumentFragment>
`;
